### DISCUSSION TITLE
List of filenames as input to apply functions to

### DISCUSSION THREAD

**Discussion by AndresKasekamp** [r-lidar/lidR#578] [Q&A]
Is it possible to input a list of filenames that functions will be applied? Similar to catalog_apply function, but the input is a list of filenames that have already been processed, and they should only be used as a buffer for unprocessed files, so I cannot completely exclude them.

**Comment by Jean-Romain** [comment]
It is unclear what you are asking

**Reply by AndresKasekamp** [reply]
For example, if I have laz files laz_1.laz, laz_2.laz, laz_3.laz, laz_4.laz and I have generated DTM tiff files from them named laz_1.tiff, laz_2.tiff, laz_3.tiff, but not laz_4.tiff. Something happened when reading in all the laz files and I could not generate the last DTM. I would need to read in all the laz files again in order to avoid edge artifacts since they are neighboring tiles. Is it possible that I don't have to regenerate laz_1.tiff, laz_2.tiff, laz_3.tiff before I get to laz_4.tiff, and I can only use the buffer from neighboring tiles.

**Reply by Jean-Romain** [reply]
Yes. If you want to restart the computation at chunk `n` use [restarting option](https://cran.r-project.org/web/packages/lidR/vignettes/lidR-LAScatalog-engine.html#error-handling). If you want to process only some tiles use [partial processing](https://cran.r-project.org/web/packages/lidR/vignettes/lidR-LAScatalog-engine.html#partial-processing).
